client/data/authorizations/resolvers.ts

```diff
@@ -3,76 +3,55 @@
  */
-import { updateAuthorizations, updateAuthorizationsSummary } from './actions';
-import { Authorization, RiskLevel } from 'wcpay/types/authorizations';
-import { Query } from '@woocommerce/navigation';
-
-export function* getAuthorizations( query: Query ): any {
-	const getMockedRows = () => {
-		const randomAmount = () => {
-			return 1400 + Math.floor( Math.random() * 5000 );
-		};
-
-		const randomDate = dateI18n(
-			'M j, Y / g:iA',
-			moment.utc( new Date() ).local().toISOString()
-		);
-
-		const randomCaptureDate = dateI18n(
-			'M j, Y / g:iA',
-			moment.utc( new Date() ).add( '7', 'days' ).local().toISOString()
-		);
-
-		const randomRisk = (): RiskLevel => {
-			const risks: Array< RiskLevel > = [ 'elevated', 'normal', 'high' ];
-
-			return risks[ Math.floor( Math.random() * risks.length ) ];
-		};
-
-		const data = Array( 10 ).fill( 0 );
-
-		return data.map( () => {
-			const dataAmount = randomAmount();
-			const dataRisk = randomRisk();
-
-			const authorization: Authorization = {
-				authorization_id: '123',
-				authorized_on: randomDate,
-				capture_by: randomCaptureDate,
-				order: {
-					number: 24,
-					customer_url: 'https://doggo.com',
-					url: 'https://doggo.com',
-				},
-				risk_level: dataRisk,
-				amount: dataAmount,
-				customer_email: 'good_boy@doge.com',
-				customer_country: 'Kingdom of Dogs',
-				customer_name: 'Good boy',
-			};
-
-			return authorization;
-		} );
-	};
-
-	yield updateAuthorizations( query, getMockedRows() );
+import {
+	updateAuthorizations,
+	updateAuthorization,
+	updateAuthorizationsSummary,
+	updateErrorForAuthorizations,
+} from './actions';
+import { Authorization, AuthorizationsList } from 'wcpay/types/authorizations';
+import { NAMESPACE } from '../constants';
+import { ApiError } from 'wcpay/types/errors';
+
+export function* getAuthorizations( query: Query ): Generator< unknown > {
+	const {
+		paged = 1,
+		per_page: perPage = 25,
+		orderby = 'created',
+		order = 'desc',
+	} = query;
+	const path = addQueryArgs( `${ NAMESPACE }/authorizations`, {
+		page: paged,
+		pagesize: perPage,
+		sort: orderby,
+		direction: order,
+	} );
+
+	try {
+		const result = yield apiFetch( { path } );
+		yield updateAuthorizations( query, result as AuthorizationsList );
+	} catch ( error ) {
+		yield updateErrorForAuthorizations( query, error as ApiError );
+	}
 }
 
-export function* getAuthorizationsSummary( query: Query ): any {
-	const data = {
-		count: 10,
-		total: 100,
-		currency: 'USD',
-		store_currencies: [ 'USD' ],
-		customer_currencies: [ 'USD', 'EUR' ],
-		totalAmount: 126790,
-	};
+export function* getAuthorization(
+	paymentIntentId: string
+): Generator< unknown > {
+	yield updateAuthorization( {
+		payment_intent_id: paymentIntentId,
+	} as Authorization );
+}
 
-	yield updateAuthorizationsSummary( query, data );
+export function* getAuthorizationsSummary(
+	query: Query
+): Generator< unknown > {
+	yield updateAuthorizationsSummary( query, {} );
```

Comment:
I think we are missing an api fetch here + error handling

Reply:
Yes, I took it from [here](https://www.typescriptlang.org/tsconfig#resolveJsonModule)

Reply:
Indeed. I'll add a TODO comment here, saying that this is implemented in this other [PR](https://github.com/Automattic/woocommerce-payments/pull/4982).

Reply:
Done here [93e59f2](https://github.com/Automattic/woocommerce-payments/pull/4961/commits/93e59f279f2017b38550cbb8995889774c13f04f)